cmd/kube-apiserver/app/options/options.go

```diff
@@ -238,4 +238,12 @@ func (s *ServerRunOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.ServiceAccountSigningKeyFile, "service-account-signing-key-file", s.ServiceAccountSigningKeyFile, ""+
 		"Path to the file that contains the current private key of the service account token issuer. The issuer will sign issued ID tokens with this private key. (Requires the 'TokenRequest' feature gate.)")
 
+	// Unhide deprecated flags. We want deprecated flags to show in kube-apiserver help.
+	// We have some hidden flags, but we might as well unhide these when they are deprecated,
+	// as silently deprecating and removing (even hidden) things is unkind to people who use them.
+	fs.VisitAll(func(flag *pflag.Flag) {
+		if len(flag.Deprecated) > 0 {
+			flag.Hidden = false
```

Comment:
Hi, will this take the hidden flag back?

What should I do if I *do* want to hide it?

Reply:
Ref https://github.com/kubernetes/kubernetes/issues/62009#issuecomment-380246409
This is intended.

Reply:
OK.
~What should I do if I do want to hide it?~

I do not want to hide it. :laughing: